x-pack/lib/monitoring/inputs/metrics.rb

```diff
@@ -211,9 +211,18 @@ def fetch_global_stats
     end
 
     def extract_cluster_uuids(stats)
-      result = stats.extract_metrics([:stats, :pipelines, :main, :config], :cluster_uuids)
-      if result && !result[:cluster_uuids].empty?
-        cluster_uuids = result[:cluster_uuids]
+      cluster_uuids = []
+      agent.running_pipelines.each do |pipeline_id, _|
+        unless pipeline_id.to_sym == :".monitoring-logstash"
+          path = [:stats, :pipelines, pipeline_id.to_sym, :config]
+          found_cluster_uuids = stats.extract_metrics(path, :cluster_uuids)
+          if found_cluster_uuids && !found_cluster_uuids[:cluster_uuids].empty?
+            cluster_uuids |= found_cluster_uuids[:cluster_uuids]
+          end
+        end
+      end
+
+      unless cluster_uuids.empty?
```

Comment:
Nitpick: I understand the desire to minimize the diff, but `unless`...`else` blocks can get pretty confusing 😩